### PR TITLE
Allow cursor and order modifiers on streams

### PR DESCRIPTION
**Pull request by cmichi**
Hi @Synesso,

this PR should implement #19. Since this is my first contact with Scala I expect that you probably have comments :-).

### CODE REVIEW

**Comment by Synesso** [comment]
Thank you @cmichi. I'm happy to get your PR. 

Can you tighten up the types of these params? I suggest using `sealed trait`s for both.

e.g. looking at the [all effects endpoint](https://www.stellar.org/developers/horizon/reference/endpoints/effects-all.html#arguments) specifically:
* order can be `Asc` or `Desc`.
* cursor can be `Epoch`, `Record(Long)` or `Now`.

**Comment by Synesso** [comment]
I just re-read about this being your first contact with Scala, so I'll expand on the sealed traits comment. You can think of them as a class hierarchy that is closed to extension.

e.g. 
```
sealed trait Order
case object Asc extends Order
case object Desc extends Order
```

**Comment by cmichi** [comment]
@Synesso Thanks for the comments! I updated the PR.

I have two questions:
1) Regarding your comment

> cursor can be Epoch, Record(Long) or Now.

I don't understand what you refer to with "Epoch"? As I understand the Stellar doc there is the possibility to either submit a number (0…n) or "now".

2) Is it possible to execute the codecov tests without pushing here?

**Comment by Synesso** [comment]
I suggested `Epoch` to cover the case where the user doesn't provide a value. But you're right, it's just an alias for `0`. So to make things simpler we can just have the numeric and `now` values.

You can do a local coverage check with `sbt coverate test it:test coverageReport`.

**Comment by cmichi** [comment]
Okay thanks! I'm not sure how to get the test coverage up, to me it seems that the tests already cover the cases which `getStream(…)` can be called with. Do you have any hints?

File: src/main/scala/stellar/sdk/inet/HorizonCursor.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package stellar.sdk.inet
+
+
+sealed trait HorizonCursor {
+  val value: Long = 0
+}
+case object Now extends HorizonCursor
+case class Record(override val value: Long) extends HorizonCursor
```
Comment (Synesso) [inline src/main/scala/stellar/sdk/inet/HorizonCursor.scala]: Better to have `value` part of `Record` only. `Now` doesn't need it. i.e.

```
sealed trait HorizonCursor
case object Now extends HorizonCursor
case class Record(value: Long) extends HorizonCursor
```

Comment (Synesso) [reply src/main/scala/stellar/sdk/inet/HorizonCursor.scala]: Now I think of it, without `Epoch` this type can simplified to just `Option[Long]`.

File: src/main/scala/stellar/sdk/inet/HorizonAccess.scala [hunk 2]
```diff
@@ -69,9 +69,19 @@ class Horizon(uri: URI,
...
+  def getStream[T: ClassTag](path: String, de: CustomSerializer[T], cursor: HorizonCursor, order: HorizonOrder, params: Map[String, String] = Map.empty)
                             (implicit ec: ExecutionContext, m: Manifest[T]): Future[Stream[T]] = {
 
+    val orderParam = order match {
+      case Asc => "asc"
+      case Desc => "desc"
+    }
+    val cursorParam = cursor match {
+      case Now => "now"
+      case Record(_) => cursor.value.toString
```
Comment (Synesso) [inline src/main/scala/stellar/sdk/inet/HorizonAccess.scala]: If `HorizonCursor` is kept, this can be `case Record(r) => r.toString`.
If simplified to `Option[Long]` (see below), this can be `cursorParam.map(_.toString).getOrElse("now")`

**Comment by Synesso** [comment]
If I click through on the coverage report above, I think it's indicating missing coverage on HorizonAccess.

**Comment by cmichi** [comment]
Regarding `HorizonCursor` and `Now`: I like the explicitness of having an enum which denotes the difference and would prefer the way it is now. Would that be okay for you or should I change it?

I can't wrap my head about the missing test coverage and tried a number of things, but none seem to work. codecov always complains about missed tests for `Desc` and `Now`, although the tests should definitely cover that (e.g. `"fetch effects descending beginning from now"` calls `network.effects(Now, Desc)`). Do you maybe have an idea?

**Comment by Synesso** [comment]
No problem with `HorizonCursor`. 👍 
I see what you mean about the coverage error. That doesn't make sense. I'm looking into it.

**Comment by Synesso** [comment]
I ran the coverage 3 times. It failed the first time and succeeded after that.

I re-ran the travis job and the coverage passed. The above PR page has not updated, but the report is https://codecov.io/github/Synesso/scala-stellar-sdk/commit/869820b9cec958f9602fbfd10b4c9bacd1a6429f

I'm going to raise an issue on the coverage project, as I can't see a good reason for the intermittent failure.

Thanks very much for your PR!

**Comment by cmichi** [comment]
Ah and btw. thanks for adding the annotated docs!
